src/lazy/encoder/writer.rs

```diff
@@ -41,14 +43,14 @@ impl EncodingContext {
 }
 
 /// An Ion writer that maintains a symbol table and creates new entries as needed.
-pub struct ApplicationWriter<E: LazyEncoder, Output: Write> {
+pub struct ApplicationWriter<E: Encoding, Output: Write> {
```

Comment:
🗺️ My goal is to eventually do enough renames that this will be either:

```rust
Writer<Text_1_1, Vec<u8>>
```
or
```rust
Writer<v1_1::Text, Vec<u8>>
```

Reply:
I like the second one better because it would allow you to have e.g. `use <...>::v1_1::*;` *i.e. you can `use` a specific Ion version) and then
```rust
Writer<Text, Vec<u8>>
```